plugins/carbon_black_cloud/icon_carbon_black_cloud/connection/connection.py

```diff
@@ -146,3 +152,68 @@ def test(self):
                 data=str(error),
             )
         return {"success": True}
+
+    def test_task(self):
+        self.logger.info("Running a connection test to Carbon Black Cloud")
+
+        alerts_endpoint = self.base_url + f"/api/alerts/v7/orgs/{self.org_key}/alerts/_search"
+        observations_endpoint = self.base_url + f"/api/investigate/v2/orgs/{self.org_key}/observations/search_jobs"
+
+        current_time = datetime.now(timezone.utc)
+        now = current_time.strftime(TIME_FORMAT)
+        minus_five_mins = (current_time - timedelta(minutes=5)).strftime(TIME_FORMAT)
+
+        fd = {
+            alerts_endpoint: {
+                "time_range": {"start": minus_five_mins, "end": now},
+                "criteria": {},
+                "start": "1",
+                "rows": str(1),  # max number of results that can be returned
+                "sort": [{"field": ALERT_TIME_FIELD, "order": "ASC"}],
+            },
+            observations_endpoint: {
+                "rows": 1,
+                "start": 0,
+                "fields": ["*"],
+                "criteria": {"observation_type": OBSERVATION_TYPES},
+                "sort": [{"field": OBSERVATION_TIME_FIELD, "order": "asc"}],
+                "time_range": {"start": minus_five_mins, "end": now},
+            },
+        }
+
+        return_msg = "Task connection test to Carbon Black Cloud successful"
+        return_msg_error = "Task connection test to Carbon Black Cloud failed"
+        self.logger.info("Testing get alerts")
+
+        failed = ""
+
+        for key, value in fd.items():
+            try:
+                # Get a 200 from the search jobs endpoint
+                self.request_api(key, value)
+
+            except HTTPErrorException as error:
+                # If we've already hit the same error (usually invalid creds)
+                # on the first one then skip, otherwise we're just repeating ourselves
+                if error.cause in failed:
+                    pass
+                else:
+                    return_msg_error += f"\n{error.cause}"
+                    return_msg_error += f"\n{error.assistance}"
+                    return_msg_error += f"\nThis URL: {key} failed "
```

Comment:
Couple of minors could just append all of this on one line? For the URL section should we change it to only mention which endpoint? Something like:

```
Task connection test to Carbon Black Cloud failed
Authentication Error. Please verify connection details are correct.
Please verify that your Secret Key and API ID values in the plugin connection are correct.
The endpoint for alerts failed.
```

and it both fail: `The endpoint for alerts and observations failed.` ?

Instead of the pass section this could be used to add the second endpoint to which variable is being used to track it and then at the bottom `if fail` create this last string of information.

Reply:
@joneill-r7 I updated the testing field in this PR to show the new results of the changes made per this suggestion ^

Reply:
thanks @cmcnally-r7 , just added a comment around this bit as it looks like it will still log the whole URL?